src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -66,90 +90,133 @@ public void setFiles(FileCollection files) {
         this.files = files;
     }
 
-    @Input
+    /**
+     * @return algorithm
+     * @deprecated Use getAlgorithmProperty() instead
+     */
+    @Deprecated
+    @Internal
     public Algorithm getAlgorithm() {
+        return algorithm.get();
+    }
+
+    @Input
+    public Property<Algorithm> getChecksumAlgorithm() {
         return algorithm;
     }
 
     public void setAlgorithm(Algorithm algorithm) {
-        this.algorithm = algorithm;
+        this.algorithm.set(algorithm);
     }
 
-    @OutputDirectory
+    /**
+     * @return output directory
+     * @deprecated Use getOutputDirProperty() instead
+     */
+    @Deprecated
+    @Internal
     public File getOutputDir() {
+        return outputDir.get().getAsFile();
+    }
+
+    @OutputDirectory
+    public DirectoryProperty getOutputDirectory() {
         return outputDir;
     }
 
-    public void setOutputDir(File outputDir) {
-        if (outputDir.exists() && !outputDir.isDirectory()) {
+    public void setOutputDir(File outputDirAsFile) {
+        if (outputDirAsFile.exists() && !outputDirAsFile.isDirectory()) {
             throw new IllegalArgumentException("Output directory must be a directory.");
```

Comment:
:question:  This check requires that the output directory exists at the time the task is configured. This doesn't make much sense to me.